optuna/_optimize.py

```diff
@@ -0,0 +1,248 @@
+import copy
+import datetime
+import gc
+import math
+from typing import Callable
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Type
+import warnings
+
+import joblib
+from joblib import delayed
+from joblib import Parallel
+
+import optuna
+from optuna import exceptions
+from optuna import logging
+from optuna import progress_bar as pbar_module
+from optuna import storages
+from optuna import trial as trial_module
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+_logger = logging.get_logger(__name__)
+
+
+def _optimize(
```

Comment:
This function is basically a copy of previous logic in `Study.optimize` with some cosmetic fixes such as reduced indirections, reducing arguments being passed around, and handling of the progress bar.